src/renderer/components/wallet/keystore/ImportKeystore.tsx

```diff
@@ -1,78 +1,88 @@
 import React, { useCallback, useState, useEffect, useMemo } from 'react'
 
-import { CheckCircleTwoTone } from '@ant-design/icons'
 import { Keystore } from '@xchainjs/xchain-crypto'
 import { Form, Spin } from 'antd'
 import { Store } from 'antd/lib/form/interface'
 import Paragraph from 'antd/lib/typography/Paragraph'
+import * as FP from 'fp-ts/lib/function'
 import * as O from 'fp-ts/lib/Option'
-import { none, Option, some } from 'fp-ts/lib/Option'
 import { useObservableState } from 'observable-hooks'
 import { useIntl } from 'react-intl'
 import { useHistory } from 'react-router-dom'
 
-import { ReactComponent as UploadIcon } from '../../../assets/svg/icon-upload-keystore.svg'
 import { useBinanceContext } from '../../../contexts/BinanceContext'
-import { useWalletContext } from '../../../contexts/WalletContext'
+import { liveData } from '../../../helpers/rx/liveData'
 import * as walletRoutes from '../../../routes/wallet'
+import { ImportKeystoreLD, LoadKeystoreLD } from '../../../services/wallet/types'
 import { Button } from '../../uielements/button'
 import { InputPassword } from '../../uielements/input'
 import * as Styled from './Keystore.styles'
 
-export const ImportKeystore: React.FC = (): JSX.Element => {
+type Props = {
+  importKeystore$: (keystore: Keystore, password: string) => ImportKeystoreLD
+  loadKeystore$: () => LoadKeystoreLD
+}
+
+export const ImportKeystore: React.FC<Props> = (props): JSX.Element => {
+  const { importKeystore$, loadKeystore$ } = props
+
   const history = useHistory()
   const [form] = Form.useForm()
 
   const intl = useIntl()
 
-  const { keystoreService } = useWalletContext()
-  const { loadKeystore } = keystoreService
-
   const { clientViewState$ } = useBinanceContext()
   const clientViewState = useObservableState(clientViewState$, 'notready')
   const [importing, setImporting] = useState(false)
-  const [importError, setImportError] = useState<Option<Error>>(none)
+  const [importError, setImportError] = useState<O.Option<Error>>(O.none)
```

Comment:
By using and subscribing to `importKeystore$` and `loadKeystore$`, we can store it's value (which is RemoteData) into a state. This state includes `pending`, `failure` and `success`. That's we can remove extra state handling of `importing`, `importError`.

To get an idea, check `Swap.tsx` component, where we deal with a value of `RemoteData` by subscribing to `swap$`. It's not the same stream or `RemoteData` here in ImportKeystore` we have to deal with, but the same idea behind.